src/components/Header/Header.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {ReactNode} from 'react'
+import {StyleSheet, TouchableOpacity} from 'react-native'
+import type {ImageStyle, StyleProp, TextStyle, ViewStyle} from 'react-native'
+import styled from 'styled-components/native'
+import {TextBold} from '../Text/Text'
+import {Images} from '../../theme'
+import {hitSlop, metrics} from '../../helpers'
+import {useTheme} from '../../hooks'
+
+export interface HeaderProps {
+  /**
+   * The top padding for the notch of device, usually we will use react-native-safe-area-context to get the inset
+   */
+  paddingTop?: number
+  /**
+   * Height of the header
+   */
+  height?: number
+  /**
+   * Style of Header container
+   */
+  containerStyle?: StyleProp<ViewStyle>
+  /**
+   * Width of Header bottom border
+   */
+  borderBottomWidth?: number
+  /**
+   * Color of Header bottom border
+   */
+  borderBottomColor?: string
+  /**
+   * Title of the header, will be use if no custom CenterComponent is defined
+   */
+  title?: string
+  /**
+   * The color of Header title
+   */
+  titleColor?: string
+  /**
+   * The custom style of Header title
+   */
+  titleStyle?: StyleProp<TextStyle>
+  /**
+   * Background color of the Header
+   */
+  backgroundColor?: string
+  /**
+   * Show the back to previous screen icon, default value is true
+   */
+  hasBackButton?: boolean
+  /**
+   * The color of back to previous screen icon
+   */
+  backIconColor?: string
+  /**
+   * The action for the back to previous screen icon
+   */
+  backButtonPress?: () => void
+  /**
+   * The left component of the Header
+   */
+  leftComponent?: ReactNode
+  /**
+   * The style of left component of the Header
+   */
+  leftStyle?: StyleProp<ViewStyle>
+  /**
+   * The center component of the Header, will replace Title
+   */
+  centerComponent?: ReactNode
+  /**
+   * The style of center component of the Header
+   */
+  centerStyle?: StyleProp<ViewStyle>
+  /**
+   * The right component of the Header
+   */
+  rightComponent?: ReactNode
+  /**
+   * The style of right component of the Header
+   */
+  rightStyle?: StyleProp<ViewStyle>
+}
+
+const Header: React.FC<HeaderProps> = ({
+  paddingTop,
+  height,
+  containerStyle,
+  borderBottomWidth,
+  borderBottomColor,
+  title,
+  titleColor,
+  titleStyle,
+  backgroundColor,
+  hasBackButton = true,
+  backIconColor,
+  backButtonPress,
+  leftComponent,
+  leftStyle,
+  centerComponent,
+  centerStyle,
+  rightComponent,
+  rightStyle,
+}) => {
+  const HeaderTheme = useTheme().components.Header
+  return (
+    <Container
+      backgroundColor={backgroundColor ?? HeaderTheme.backgroundColor}
+      height={(height ?? (HeaderTheme.height as number)) + (paddingTop ?? (HeaderTheme.paddingTop as number))}
+      paddingTop={paddingTop ?? HeaderTheme.paddingTop}
```

Comment:
The `HeaderTheme.paddingTop` is metrics.xxl, you don't use the insets from `react-native-safe-area-context`

Reply:
That is because I don't want to add `react-native-safe-area-context` to our package. I mentioned it in the comment of the props above
<img width="838" alt="Screenshot 2023-11-09 at 10 26 21" src="https://github.com/saigontechnology/rn-base-component/assets/114563576/d7fea44d-e044-4bf3-8127-819bcf95bd8c">

Reply:
My purpose was something like this
![Screenshot 2023-11-09 at 11 31 36](https://github.com/saigontechnology/rn-base-component/assets/114563576/bcddb969-d70c-44c5-8250-8c45d22acfd1)